source/firstprogram.rst

```diff
@@ -1,402 +1,621 @@
-==========================
+Writing your First Program:
+===========================
 
-This section is a detailed guide on how to build a `SIR Model <link for SIR URL>`_ in BharatSim from scratch. By the end of this section, you should be able to write and execute a SIR model by yourself.
+This section is a detailed guide for a novice user on how to build a `SIR Model <https://bharatsim.readthedocs.io/en/latest/epidemiology-sir-model.html>`_ in BharatSim from scratch. By the end of this section, you should be able to write and execute a SIR model by yourself.
 
-Any model built using this framework contains different classes which are essentially extensions of different `Nodes <#>`_. To properly build a SIR model from scratch, you will first need to define these classes and the properties associated with them. These classes are grouped under different components of the model. Here, the SIR model has two different components:-
-1. `Agent <#>`_
-2. `Network <#>`_
+Any model built on this framework contains various classes which are essentially extensions of different Nodes. To build a SIR model from scratch, one needs to define the said classes and the properties and relationships associated with them. In the basic SIR there are two main components:
 
-These components are related to each other via simple relations which are described in detail later.
+1. `Agent`
+2. `Network`
 
-Setting up the components of the Network
-^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Agents are simply the individuals in the population and the Network is the manner in each of these individuals interact with one another. 
+However, to begin writing this program we assume that all the individuals are present in one location, hence neglecting the Network component. 
 
-The Network class gives a sense of geography to the model. The different components of the Network are simple locations like houses, offices, schools etc. The agents move about between these locations as shown in the figure below.
+Creating an Empty Class:
+^^^^^^^^^^^^^^^^^^^^^^^^
+Before we get to the actual coding aspect of the code, it is essential that we set up an empty and make sure that the framework is not giving any errors. 
+Navigate to ``BharatSim/src/main/scala/com/bharatsim/examples/epidemiology`` in IntelliJ or VSCode and right click on the folder. Upon right clicking, select the New Package option and name it SIR_Scratch. An empty folder should appear named SIR_Scratch and create a new scala class called Main. A yellow circular icon should appear. 
 
-The first step to building your model is to create locations which are part of the Network, by creating a new scala `class <https://docs.scala-lang.org/tour/classes.htm>`_. You can create a new class for adding houses to the network, in the following way:-
+The empty class should look like this. 
 
-1. Since ``House`` is a component of the Network, you have to import the Network class.
+.. image:: _static/images/New_class.png

```

Comment:
You have not added these files to git! So they haven't been uploaded. Add all your images using `git add` before pushing them to the repo.

Reply:
This is done, so resolving.